### PR TITLE
Added filtration to controller "Registration"

### PR DESCRIPTION
**Pull request by Grimooar**
*Added propertu status in Appsettings 
*aplying filter in controller

### CODE REVIEW

File: src/Kirel.Identity.Server.API/appsettings.json [hunk 1]
```diff
@@ -5,6 +5,9 @@
       "Microsoft.AspNetCore": "Warning"
     }
   },
+  "ControllerState": {
```
Comment (Insei) [inline src/Kirel.Identity.Server.API/appsettings.json]: I think, we should rename this property name, and use something like:
"RegistrationEnable":  true/false, without info about controllers and other details of the implementation.

File: src/Kirel.Identity.Server.Core/Extensions/ServicesExtension.cs [hunk 2]
```diff
@@ -15,13 +15,13 @@ public static class ServicesExtension
     /// </summary>
     /// <param name="services"> services collection </param>
     /// <param name="jwtConfig"> JWT Token generation config </param>
-    public static void AddServices(this IServiceCollection services, JwtAuthenticationConfig jwtConfig)
+    public static void AddServices(this IServiceCollection services, JwtAuthenticationConfig jwtConfig  )
```
Comment (Insei) [inline src/Kirel.Identity.Server.Core/Extensions/ServicesExtension.cs:18]: thy you add space here? i think it's not needed

File: src/Kirel.Identity.Server.Core/Filters/EnableIfSettingFilter.cs [hunk 3]
```diff
@@ -0,0 +1,43 @@
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Filters;
+
+namespace Kirel.Identity.Server.Core.Filters;
+
+/// <summary>
+/// Custom attribute for enabling or disabling action execution based on a specified setting.
+/// </summary>
+[AttributeUsage(AttributeTargets.Class | AttributeTargets.Method)]
+public class EnableIfSettingAttribute : Attribute, IActionFilter
```
Comment (Insei) [inline src/Kirel.Identity.Server.Core/Filters/EnableIfSettingFilter.cs]: Why not simply EnabledControllerAttribute? I think this name will be prefer.

File: src/Kirel.Identity.Server.API/Controllers/RegistrationController.cs [hunk 4]
```diff
@@ -7,8 +8,11 @@
 namespace Kirel.Identity.Server.API.Controllers;
 
 /// <inheritdoc />
+
 [ApiController]
 [Route("registration")]
+[EnableIfSetting(true)] 
+[ApiExplorerSettings(IgnoreApi = true)]
```
Comment (Insei) [inline src/Kirel.Identity.Server.API/Controllers/RegistrationController.cs]: Why we ignore this controller always in swagger? It's not correct, if we enable this controller in settings, we should show swagger documentation for him.

**Review by Insei — CHANGES_REQUESTED** [review]

**Comment by Grimooar** [comment]
fully working version without any issues

**Comment by Insei** [comment]
Closed, we have a new one: https://github.com/Kirel-tech/Kirel.Identity/pull/35

**Pull request closed without merging**